src/Server/GRPCServer.cpp

```diff
@@ -990,41 +989,10 @@ namespace
         assert(!pipeline);
         auto source = query_context->getInputFormat(
             input_format, *read_buffer, header, query_context->getSettings().max_insert_block_size);
+
         QueryPipelineBuilder builder;
         builder.init(Pipe(source));
 
-        /// Add default values if necessary.
-        if (ast)
-        {
-            if (insert_query)
-            {
-                auto table_id = StorageID::createEmpty();
-
-                if (insert_query->table_id)
-                {
-                    table_id = query_context->resolveStorageID(insert_query->table_id, Context::ResolveOrdinary);
-                }
-                else
-                {
-                    StorageID local_table_id(insert_query->getDatabase(), insert_query->getTable());
-                    table_id = query_context->resolveStorageID(local_table_id, Context::ResolveOrdinary);
```

Comment:
Here was the bug, because `Context::ResolveOrdinary` doesn't include `Context::ResolveExternal`, i.e. it doesn't resolve temporary and external tables. So to fix the bug it seemed it was enough just to replace `Context::ResolveOrdinary` with `Context::ResolveAll`.

But then I found that this code was probably copied from [InputStreamFromASTInsertQuery.cpp](https://github.com/ClickHouse/ClickHouse/blob/aeac8cb6212ce3658a5ffb99116ddedf3910ac65/src/DataStreams/InputStreamFromASTInsertQuery.cpp#L59-L65) and that file was removed with DataStreams. And now there is no such code for other handlers (tcp, http). Anyway for gRPC we have test [test_grpc_protocol/test.py::test_insert_default_column](https://github.com/ClickHouse/ClickHouse/blob/b2ba0c4320044bca6c98aa304450c869843b6c8a/tests/integration/test_grpc_protocol/test.py#L179) which checks that gRPC can add default columns while inserting.

Reply:
@KochetovNicolai Сan you check whether `AddingDefaultsTransform` was necessary here?